mesonbuild/compilers/mixins/gnu.py

```diff
@@ -92,6 +92,275 @@
+    ],
+    "4.3.0": [
+        "-Wclobbered",
+        "-Warray-bounds=2",
+        "-Wempty-body",
+        "-Wignored-qualifiers",
+        "-Wlogical-op",
+        "-Wstrict-aliasing=3",
+        "-Wtype-limits",
+        "-Wvla",
+    ],
+    "4.4.0": [
+        "-Wpacked-bitfield-compat",
+    ],
+    "4.6.0": [
+        "-Wdouble-promotion",
+        "-Wsuggest-attribute=const",
+        "-Wsuggest-attribute=noreturn",
+        "-Wsuggest-attribute=pure",
+        "-Wtrampolines",
+    ],
+    "4.7.0": [
+        "-Wvector-operation-performance",
+    ],
+    "4.8.0": [
+        "-Wsuggest-attribute=format",
+    ],
+    "4.9.0": [
+        "-Wdate-time",
+    ],
+    "5.1.0": [
+        "-Wformat-signedness",
+        "-Wnormalized=nfc",
+    ],
+    "6.1.0": [
+        "-Wduplicated-cond",
+        "-Wnull-dereference",
+        "-Wshift-negative-value",
+        "-Wshift-overflow=2",
+        "-Wunused-const-variable=2",
+    ],
+    "7.1.0": [
+        "-Walloca",
+        "-Walloc-zero",
+        "-Wformat-overflow=2",
+        "-Wformat-truncation=2",
+        "-Wimplicit-fallthrough=3",
+        "-Wstringop-overflow=3",
+    ],
+    "7.2.0": [
+        "-Wduplicated-branches",
+    ],
+    "8.1.0": [
+        "-Wattribute-alias=2",
+        "-Wcast-align=strict",
+        "-Wcast-function-type",
+        "-Wsuggest-attribute=cold",
+        "-Wsuggest-attribute=malloc",
+    ],
+    "10.1.0": [
+        "-Wanalyzer-too-complex",
+        "-Warith-conversion",
+        "-Wstring-compare",
+    ],
+    "12.1.0": [
+        "-Wbidi-chars=ucn",
+        "-Wopenacc-parallelism",
+        "-Wtrivial-auto-var-init",
+    ],
+}  # type: T.Dict[str, T.List[str]]
+
+# GCC warnings for C
+# Omitted non-general or legacy warnings:
+#   -Wc11-c2x-compat
+#   -Wc90-c99-compat
+#   -Wc99-c11-compat
+#   -Wdeclaration-after-statement
+#   -Wtraditional
+#   -Wtraditional-conversion
+gnu_c_warning_args = {
+    "0.0.0": [
+        "-Wbad-function-cast",
+        "-Wmissing-prototypes",
+        "-Wnested-externs",
+        "-Wsign-compare",
+        "-Wstrict-prototypes",
+    ],
+    "3.4": [
+        "-Wold-style-definition",
+    ],
+    "4.1.0": [
+        "-Wc++-compat",
+    ],
+    "4.2.0": [
+        "-Woverride-init",
+    ],
+    "4.3.0": [
+        "-Wmissing-parameter-type",
+        "-Wold-style-declaration",
+    ],
+    "4.5.0": [
+        "-Wjump-misses-init",
+        "-Wunsuffixed-float-constants",
+    ],
+    "8.1.0": [
+        "-Wenum-conversion",
+    ],
+    "9.1.0": [
+        "-Wabsolute-value",
+    ],
+}  # type: T.Dict[str, T.List[str]]
+
+# GCC warnings for C++
+# Omitted non-general or legacy warnings:
+#   -Wc++0x-compat
+#   -Wc++1z-compat
+#   -Wc++2a-compat
+#   -Wctad-maybe-unsupported
+#   -Wnamespaces
+#   -Wtemplates
+gnu_cpp_warning_args = {
+    "0.0.0": [
+        "-Wctor-dtor-privacy",
+        "-Wdeprecated",
+        "-Weffc++",
+        "-Wimport",
+        "-Wnon-virtual-dtor",
```

Comment:
I think I am going to leave this one on purpose.  It's the only weird case like this, but since `-Weffc++` is so widely turned off, I think omitting it would mean people accidentally disable this (more useful and less sketchy) warning.